src/zc/form/browser/combinationwidget.py

```diff
@@ -113,7 +113,7 @@ def render(self, value):
 
     def _set_values_on_widgets(self, values):
         hasInput = self.hasInput()
-        for w, v in map(None, self.widgets, values):
+        for w, v in zip(self.widgets, values):
```

Comment:
Huh.  This is a behavioral change, when len(self.widgets) != len(values):
```
>>> map(None, 'abc', '12')
[('a', '1'), ('b', '2'), ('c', None)]
>>> zip('abc', '12')
[('a', '1'), ('b', '2')]
```

Since `map(None, ...)` is such an awkward expression, it must've been used here for a reason.

Reply:
This is a good point. In this case in line 102 an assertion for the equal length is performed. So in this case only a missing_value or the correct length should be passed `_set_values_on_widgets()`. But one could even argue, that before the refactoring there might be a possibility to get this wrong. In my opinion the last PR fixed this bug by actually setting the missing value in those cases. 

From the other perspective, a `None` for a widget would result in an `AttributeError` in line 117. A `None` as a value passed to `w.setRenderedValue(v)` might result in an `AttributeError` later on, as an already rendered data is expected. (e.g. here https://github.com/zopefoundation/zope.formlib/blob/master/src/zope/formlib/widget.py#L551) So for me it seems reasonable to keep it as zip. Even if there had been a reason, this leads to errors now.

Reply:
Thank you for checking!

I'm fine with `zip()`, if the lengths are known to be equal.